### DISCUSSION TITLE
Porting to v5, issue with FromForm

### DISCUSSION THREAD

**Discussion by InstinctBas** [rwf2/Rocket#2487] [Questions] [answered]
Finally diving into converting our code to 5.0 and running into a weird issue with FromForm

I've got a form like this:
```
#[derive(FromForm, Serialize)]
pub struct MyForm {
	name: String,
	type: Option<String>,
	include_something: Option<bool>,
}
```

Which is used as:
```
#[get("/?<search..>")]
fn list(search: Form<MyForm>) -> ...
```

I'm now getting a compilation error like this:
```
the trait `FromFormField<'_>` is not implemented for `rocket::form::Form<MyForm>`
```

I've tried adding lifetime thingies to it but that didn't help. Seeing it's complaining about `FromFormField` I'm guessing one of the fields in my struct is not supported?

Any suggestions what I should change?

**Comment by robertwayne** [comment, marked as answer]
You can remove the wrapper Form\<T> from your parameters in v0.5: see __[here](https://rocket.rs/v0.5-rc/guide/upgrading/#queries-as-forms)__.

```rs
#[get("/?<search..>")]
fn list(search: MyForm) -> {}
```

**Reply by InstinctBas** [reply]
Brilliant! That is an amazing improvement, could probably use an example showing this off in the documentation where its discussing forms...

**Reply by InstinctBas** [reply]
Which you're linking and now I'm scratching my head why I missed it..
